fdm-calculator/src/balance/nitrogen/supply/mineralization.ts

```diff
@@ -0,0 +1,117 @@
+import type { fdmSchema } from "@svenvw/fdm-core"
+import type {
+    NitrogenSupplyMineralization,
+    SoilAnalysisPicked,
+    NitrogenBalanceInput,
+} from "../types"
+import Decimal from "decimal.js"
+import { differenceInCalendarDays } from "date-fns"
+
+/**
+ * Calculates the amount of nitrogen supplied through soil mineralization using Minip.
+ *
+ * This function determines the mineralization based on the soil analyses conducted.
+ * @param soilAnalysis - Combined soil analysis data for the field.
+ * @param timeFrame - The timeframe for which to calculate the nitrogen mineralization.
+ * @returns The NitrogenSupplyMineralization object containing the total amount of Nitrogen mineralized.
+ */
+export function calculateNitrogenSupplyBySoilMineralization(
+    soilAnalysis: SoilAnalysisPicked,
+    timeFrame: NitrogenBalanceInput["timeFrame"],
+): NitrogenSupplyMineralization {
+    let mineralizationValue =
+        calculateNitrogenSupplyBySoilMineralizationUsingMinip(
+            soilAnalysis.a_c_of,
+            soilAnalysis.a_cn_fr,
+            soilAnalysis.a_density_sa,
+        )
+
+    // Limit the min and max values for the mineralization
+    if (mineralizationValue.greaterThan(250)) {
+        mineralizationValue = new Decimal(250)
+    }
+    if (mineralizationValue.lessThan(5)) {
+        mineralizationValue = new Decimal(5)
+    }
+
+    // Adjust for the number of days
+    const timeFrameDays = new Decimal(
+        differenceInCalendarDays(timeFrame.end, timeFrame.start),
+    )
+    // Ensure timeFrameDays is positive
+    if (timeFrameDays.lessThanOrEqualTo(0)) {
+        return { total: new Decimal(0) }
```

Comment:
check: if you set to zero, this does not lead to division by zero later?

Reply:
No at line 45 1 is added, because `differenceInCalendarDays` will return 0 if start and end are the same and if start is 2025-01-01 and end is 2025-01-02 it will give 1.